src/components/dev-dot-content/mdx-components/mdx-alert/variants.tsx

```diff
@@ -0,0 +1,23 @@
+import { MdxInlineAlert } from '.'
+import { MdxInlineAlertProps } from './types'
+
+/**
+ * Variants for composed authoring primitives.
+ * Where the type doesn't need to be set.
+ */
+
+export function MdxTip(props: Omit<MdxInlineAlertProps, 'type'>) {
+	return <MdxInlineAlert type="tip" {...props} />
+}
+
+export function MdxHighlight(props: Omit<MdxInlineAlertProps, 'type'>) {
+	return <MdxInlineAlert type="highlight" {...props} />
+}
+
+export function MdxNote(props: Omit<MdxInlineAlertProps, 'type'>) {
+	return <MdxInlineAlert type="note" {...props} />
+}
+
+export function MdxWarning(props: Omit<MdxInlineAlertProps, 'type'>) {
+	return <MdxInlineAlert type="warning" {...props} />
+}
```

Comment:
💭 I think we should set `type` after spreading `props` for each of these, that way it can't be accidentally overwritten in MDX (where the TS checks can't happen).

```suggestion
export function MdxTip(props: Omit<MdxInlineAlertProps, 'type'>) {
	return <MdxInlineAlert {...props} type="tip" />
}

export function MdxHighlight(props: Omit<MdxInlineAlertProps, 'type'>) {
	return <MdxInlineAlert {...props} type="highlight" />
}

export function MdxNote(props: Omit<MdxInlineAlertProps, 'type'>) {
	return <MdxInlineAlert {...props} type="note" />
}

export function MdxWarning(props: Omit<MdxInlineAlertProps, 'type'>) {
	return <MdxInlineAlert {...props} type="warning" />
}
```